superset/migrations/versions/8d49a37823bf_remove_tables_uniqueness.py

```diff
@@ -0,0 +1,79 @@
+"""remove_tables_uniqueness
+
+Licensed to the Apache Software Foundation (ASF) under one or more
+contributor license agreements.  See the NOTICE file distributed with
+this work for additional information regarding copyright ownership.
+The ASF licenses this file to You under the Apache License, Version 2.0
+(the "License"); you may not use this file except in compliance with
+the License.  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+Revision ID: 8d49a37823bf
+Revises: 18dc26817ad2
+Create Date: 2019-01-20 11:44:14.640628
+
+"""
+
+# revision identifiers, used by Alembic.
+revision = '8d49a37823bf'
+down_revision = '18dc26817ad2'
+
+from alembic import op
+import sqlalchemy as sa
+import pprint
+
+from superset.utils.core import generic_find_uq_constraint_name
+from collections import OrderedDict
+
+def is_unique_constraint(constraint):
+    return constraint and isinstance(constraint, sa.UniqueConstraint)
+
+def is_sqlite():
+    bind = op.get_bind()
+    print(f'The bind name is {bind.dialect.name}')
+    return bind and bind.dialect and bind.dialect.name and bind.dialect.name.startswith('sqlite')
+
+def upgrade():
+    bind = op.get_bind()
+    insp = sa.engine.reflection.Inspector.from_engine(bind)
+    constraints = insp.get_unique_constraints('tables')
+    print(f'Printing the existing uniq constraints in table {constraints}')
+    if not constraints:
+        constraints = []
+    if is_sqlite():
+        existing_table = sa.Table(
+            'tables', sa.MetaData(),
+            autoload=True,
+            autoload_with=op.get_bind())
+        existing_table.constraints = set([c for c in existing_table.constraints if not is_unique_constraint(c)])
+        print(f'Doing sqlite bulk transformation with constraints {pprint.pformat(existing_table.constraints)}')
+        # We don't want to preserve the existing table_args for the tables table
+        with op.batch_alter_table('tables', copy_from=existing_table, recreate="always") as batch_op:
+            pass
+    else:
+        print(f'Doing non sqlite transformation')
+        for c in constraints:
```

Comment:
The reason I am removing all the table unique constraints is that in certain databases like postgresql, the uniq-constraint is on <database-id, schema, table> (caused by an old migration b4456560d4f3_change_table_unique_constraint.py), but that very same migration errors out in mysql, wherein the constraint is on <table-name>. Whereas the constraint is on <database-id, table_name> in sqlite. 

As you said, alembic migrations are a royal shitshow and frankly I think this is not scaleable :(. 

Anyway, at least the DB migration aspect of this diff works for all database types.